File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.java
import java.util.Arrays;
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int[] merge=new int[nums1.length+nums2.length];
        int k=0;
        double res;
        for(int i=0;i<nums1.length;i++){
           
           merge[k]=nums1[i];
           k++;
        
        }
        for(int i=0;i<nums2.length;i++){
           
           merge[k]=nums2[i];
           k++;
        
        }
        int l=merge.length;
        Arrays.sort(merge);
        if(l%2==0){
            res=(double)merge[l/2]+merge[(l/2)-1];
            res=res/2;

        }
        else
        res=(double)merge[l/2];

        return (double)res;


    }
}

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        ans=[]
        i=0
        j=0
        tl=len(nums1)+len(nums2)
        m1=m2=-1
        if tl&1==0:
            m1=tl//2
            m2=m1-1
        else:
            m1=m2=(tl//2)
        while i<len(nums1) and j<(len(nums2)):
            if nums1[i]<=nums2[j]:
                ans.append(nums1[i])
                i+=1
            else:
                ans.append(nums2[j])
                j+=1
        ans+=nums1[i:]+nums2[j:]
        return (ans[m1]+ans[m2])/2
